Result/IndicesResult.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Result;
+
+/**
+ * This class is able to iterate over raw result.
+ */
+class IndicesResult
+{
+    /**
+     * @var array
+     */
+    private $rawData;
+
+    /**
+     * @param array $rawData
+     */
+    public function __construct(array $rawData)
+    {
+        $this->rawData = $rawData;
+    }
+
+    /**
+     * @return array
+     */
+    public function getTotal()
+    {
+        return $this->extract(func_get_args(), 'total');
+    }
+
+    /**
+     * @return array
+     */
+    public function getFailed()
+    {
+        return $this->extract(func_get_args(), 'failed');
+    }
+
+    /**
+     * @return array
+     */
+    public function getSuccessful()
+    {
+        return $this->extract(func_get_args(), 'successful');
+    }
+
+    /**
+     * Extracts data from response.
+     *
+     * @param mixed  $indices
+     * @param string $name
+     *
+     * @return array
+     */
+    private function extract($indices, $name)
+    {
+        $results = [];
+        if ($indices) {
+            foreach ($indices as $index) {
+                $results[$index] = $this->rawData['_indices'][$index]['_shards'][$name];
```

Comment:
If `$index` key would not exist, we would get an error. try to filter like this:
`array_intersect_key($this->rawData['_indices'], array_flip($indices))`